File: fixtures/scale/scala/large/src/cryptobench/ciphers/djb2/Djb2Hash.scala
package cryptobench.ciphers.djb2

import cryptobench.core.HashFunction
import cryptobench.util.Bytes

/** Bernstein's hash: state * 33 + byte. */
final class Djb2Hash extends HashFunction {

  override def name: String = "djb2"

  override def digest(input: String): Long = {
    var state = 5381L
    for (b <- Bytes.of(input)) {
      state = ((state << 5) + state) + (b.toLong & 0xFF)
    }
    state
  }
}

File: fixtures/scale/java/large/src/cryptobench/ciphers/djb2/Djb2Hash.java
package cryptobench.ciphers.djb2;

import cryptobench.core.HashFunction;
import cryptobench.util.Bytes;

/** Bernstein's hash: state * 33 + byte. */
public class Djb2Hash implements HashFunction {
    @Override
    public String name() {
        return "djb2";
    }

    @Override
    public long digest(String input) {
        long state = 5381L;
        for (byte b : Bytes.of(input)) {
            state = ((state << 5) + state) + (b & 0xFF);
        }
        return state;
    }
}
